### ISSUE TITLE
Make all solid infill monotonic when Top surface pattern is Monotonic Line.

### ISSUE THREAD

**Issue by gary-wlg** [bambulab/BambuStudio#1953]
The top surface can have poor quality when using Monotonic Lines as the previous layers show through since it is not done as monotonic.

If you select Monotonic as the Top source pattern, then all solid layers below it are also printed as Monotonic. This should also be done for Monotonic Lines, as right now if you select Monotonic Lines, the solid layers below it are not printed Monotonic.

Visible scaring shows through on dark filaments.

<img width="1538" alt="Screenshot 2023-06-24 at 6 21 43 PM" src="https://github.com/bambulab/BambuStudio/assets/29345021/adee9321-8b76-497c-8e5f-19c52b77ca7d">

**Comment by QingZhangBambu** [comment]
good point

**Comment by Wolveraptor** [comment]
I do not have a fix for this however coming in as someone new to 3D Printing, I had seen this in nearly all of my prints since I received my Bambu Lab H2D. I dried my filaments, tried different silica beads, tried different drying-dehumidifying temperatures and times, watched many YouTube videos and ready many articles and performed custom filament calibrations, and I could not get rid of this. I was beginning to become frustrated and thought something was wrong with the printer.

It was not until I stumbled across this issue and when I made by **`Top Surface Pattern`**, **`Bottom Surface Pattern`**, and **`Internal Solid Infill`** pattern match, **did this go away; thank you @gary-wlg !** (I also did not know this was called `Scaring` and was Googling for everything.) **Seriously, @gary-wlg , thank you for posting this, you saved my sanity, wallet, and filament.**

**All this to say, @QingZhangBambu , if you need me to perform any testing on potential builds that contain this feature, let me know and I am happy to test. As someone new coming into 3D printing, from an end user perspective, this was frustrating to see on top surface on prints. As @gary-wlg pointed out, this would be a great feature to have and may make the new 3D printing user experience better, more `click and go`, and improve the Bambu experience overall. (Ever since discovering this my print quality has improved along with my sanity; thanks again, @gary-wlg !)**

**Comment by Gask44** [comment]
I agree with this request, but I think it could be taken one step further.
Instead of applying Top Surface settings only to the final visible layer, Bambu Studio could allow the user to choose how many of the final solid layers use the Top Surface quality settings — for example 2 or 3 layers.
Ideally, the direction could alternate between them:
Layer -2: 0°
Layer -1: 90°
Top layer: 0°
This would give the final Top Surface a much more uniform foundation. The layers directly underneath have a significant effect on the appearance of the final surface, especially on large flat parts.
This is not the same as simply increasing Top Shell Layers. The idea is to let the last 2–3 solid layers actually use the Top Surface speed, acceleration, line width and other relevant quality parameters.
I think this could noticeably improve top surface quality without requiring ironing.
